File: API/Models/Challenge.cs
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System;

namespace QuizAppApi.Models
{
    public class Challenge : BaseEntity
    {
        public string Title { get; set; }
        public int QuestionAmount { get; set; }
        [ForeignKey("QuizTypeId")]
        public QuizType QuizType { get; set; }
        public Guid QuizTypeId { get; set; }
        [ForeignKey("ColorId")]
        public Color Color { get; set; }
        public Guid ColorId { get; set; }
        public List<ChallengeCategory> CategoryList { get; set; }
    }
}

File: src/app/models/challenge.ts
import { QuizType } from './quiztype';
import { Color } from './color';
import { Category } from './category';
import { ChallengeCategory } from './challengecategory';
import { environment } from '../../environments/environment';


export class Challenge {
    constructor() {
        this.id = environment.emptyGuid;
        this.colorId = environment.emptyGuid;
        this.quizTypeId = environment.emptyGuid;
    }
    id: string;
    title: string;
    questionAmount: number;
    color: Color;
    colorId: string;
    quizType: QuizType;
    quizTypeId: string;
    creationDate: Date;
    deletionDate: Date;
    isDeleted: boolean;
    categoryList: ChallengeCategory[];
}
